Security/Authentication/Authenticator/OAuthAuthenticator.php

```diff
@@ -149,9 +149,11 @@ public function createAuthenticatedToken(PassportInterface $passport, string $fi
         }
 
         $token = new OAuthToken($credentials->getRoles($user));
-        $token->setAuthenticated(true);
         $token->setToken($credentials->getTokenString());
         $token->setUser($user);
+        if (method_exists($token, 'setAuthenticated')) {
```

Comment:
This is a headache.  If the user is set on the token, authenticated HAS to be set to true, or it will throw errors.
If user is set and isAuthenticated is false, an error is thrown in
vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 85:
$this->tokenStorage->setToken($token = $this->authenticationManager->authenticate($token));

authenticationManager is null, which is correct:
```
        if ($accessDecisionManager instanceof AuthenticationManagerInterface) {
            trigger_deprecation('symfony/security-core', '5.4', 'The $autenticationManager argument of "%s" is deprecated.', __METHOD__);
```

If you look at other token managers such as vendor/symfony/security-core/Authentication/Token/UsernamePasswordToken.php, they are calling setAuthenticated as well, with the flag to silence the deprecation.   Presumably, they will all get removed at the same time, but until then this is the only way to make it compatible with 5.4 AND 6.0.  I am putting a note in the 6.0 upgrade t to revisit this

Reply:
Figured this out.  Symfony 6 calls createToken, Symfony 5 calls createAuthenticatedToken.